apps/application-system/api/src/app/modules/application/files/file.service.ts

```diff
@@ -40,97 +46,141 @@ export class FileService {
+    type: PdfTypes,
+  ) {
+    const bucket = this.config.presignBucket
+    if (!bucket) {
+      throw new Error(' Bucket not configured.')
     }
 
-    const expiry = answers.expiry as string
+    await this.signingService
+      .getSignedDocument(DokobitFileName[type], documentToken)
+      .then((file) => {
+        const s3FileName = `${BucketTypePrefix[type]}/${application.id}.pdf`
+        this.awsService.uploadFile(
+          Buffer.from(file, 'binary'),
+          bucket,
+          s3FileName,
+        )
+      })
+  }
+
+  async requestFileSignature(
+    application: Application,
+    type: PdfTypes,
+  ): Promise<SigningServiceResponse> {
+    const answers = application.answers as FormValue
+    const externalData = application.externalData as FormValue
 
-    return { parentA, parentB, childrenAppliedFor, expiry }
+    switch (type) {
+      case PdfTypes.CHILDREN_RESIDENCE_CHANGE: {
+        const { phoneNumber, name } = applicantData(answers, externalData)
+        return await this.handleChildrenResidenceChangeSignature(
+          type,
+          application.id,
+          name,
+          phoneNumber,
+        )
+      }
+    }
   }
 
-  private async createResidenceChangePdf(
-    parentA: ParentResidenceChange,
-    parentB: ParentResidenceChange,
-    childrenAppliedFor: Array<PersonResidenceChange>,
-    expiry: string,
+  private async handleChildrenResidenceChangeSignature(
+    type: PdfTypes,
     applicationId: string,
-  ): Promise<string> {
-    const pdfBuffer = await generateResidenceChangePdf(
-      childrenAppliedFor,
-      parentA,
-      parentB,
-      expiry,
-    )
+    applicantName: string,
+    phoneNumber?: string,
+  ): Promise<SigningServiceResponse> {
+    const bucket = this.config.presignBucket
+    if (!bucket) {
+      throw new Error(' Bucket not configured.')
+    }
+
+    const s3FileName = `${BucketTypePrefix[type]}/${applicationId}.pdf`
+    const s3File = await this.awsService.getFile(bucket, s3FileName)
+    const fileContent = s3File.Body?.toString('binary')
 
-    const fileName = `children-residence-change/${parentA.ssn}/${applicationId}.pdf`
-    const bucket = this.config.presignBucket || ''
+    if (!fileContent || !phoneNumber) {
+      throw new NotFoundException(`Variables for document signing not found`)
+    }
 
-    return await this.getPresignedUrl(pdfBuffer, bucket, fileName)
+    return await this.signingService.requestSignature(
+      phoneNumber,
+      'Lögheimilisbreyting barns',
+      applicantName,
+      'Ísland',
+      DokobitFileName[type],
+      fileContent,
+    )
   }
 
-  private async getPresignedUrl(
-    buffer: Buffer,
-    bucket: string,
-    fileName: string,
-  ): Promise<string> {
-    const uploadParams = {
-      Bucket: bucket,
-      Key: fileName,
-      ContentEncoding: 'base64',
-      ContentDisposition: 'inline',
-      ContentType: 'application/pdf',
-      Body: buffer,
+  validateApplicationType(applicationType: string) {
```

Comment:
This is a very valid point, my feeling is iterate when we have the second pdf contract. Since then we have a more mature vision of this.
My current view of this and I hoped that this reflected that each applicaiton can only create a single pdf and then the pdf type enum should only have values that are reflected in the application type enum and you need to add to the pdf enum to be able to have a pdf from that applicaiton. 
I feel that it will be very valuable to get your input once the next pdf is created and if we will create that then I will make sure you are requested for that review.
Resolving this for now and we will iterate once we get there :)